src/main/java/seedu/module/logic/commands/ViewCommand.java

```diff
@@ -28,30 +30,31 @@
     private final String moduleCode;
 
     public ViewCommand(String moduleCode) {
-        this.moduleCode = moduleCode.toUpperCase();
+        this.moduleCode = moduleCode;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        Predicate<Module> findModulePredicate = module -> module.getModuleCode().equals(moduleCode);
+        SameModuleCodePredicate findModulePredicate = new SameModuleCodePredicate(moduleCode);
 
         model.updateFilteredArchivedModuleList(Model.PREDICATE_SHOW_ALL_MODULES);
 
-        Module toDisplay = model.getFilteredModuleList().stream()
-            .filter(findModulePredicate)
-            .map(module -> (Module) module)
-            .findFirst()
-            .or(() -> model.getFilteredArchivedModuleList().stream()
-                .filter(findModulePredicate)
-                .findFirst())
-            .orElseThrow(()
+        Module toDisplay;
+
+        Optional<TrackedModule> trackedModule = model.findTrackedModule(findModulePredicate);
+
+        if (!trackedModule.isPresent()) {
+            toDisplay = model.findArchivedModule(findModulePredicate).orElseThrow(()
                 -> new CommandException(MESSAGE_MODULE_NOT_FOUND));
+        } else {
+            toDisplay = trackedModule.get();
+        }
```

Comment:
I think the following implementation is cleaner.

```java
Module toDisplay = model.findTrackedModule(findModulePredicate)
  .or(() -> model.findArchivedModule(findModulePredicate)
  .orElseThrow()
```

Reply:
Ok, sure thanks.